tests/smoke/alp_zheevd.cpp

```diff
@@ -0,0 +1,352 @@
+#include "../utils/print_alp_containers.hpp"
+
+//once TEMPDISABLE is removed the code should be in the final version
+#define TEMPDISABLE
+
+using namespace alp;
+
+using BaseScalarType = double;
+using Orthogonal = structures::Orthogonal;
+
+#ifdef _COMPLEX
+using ScalarType = std::complex< BaseScalarType >;
+//not fully implemented structures
+using HermitianOrSymmetricTridiagonal = structures::HermitianTridiagonal;
+using HermitianOrSymmetric = structures::Hermitian;
+#else
+using ScalarType = BaseScalarType;
+using HermitianOrSymmetricTridiagonal = structures::SymmetricTridiagonal;
+//fully implemented structures
+using HermitianOrSymmetric = structures::Symmetric;
+#endif
+
+constexpr BaseScalarType tol = 1.e-10;
+constexpr size_t RNDSEED = 1;
+
+//temp function untill Hermitian containter is implemented
+//** gnerate symmetric-hermitian matrix in a square container */
+template<
+	typename T
+>
+std::vector< T > generate_symmherm_matrix_data(
+	size_t N,
+	const typename std::enable_if<
+		grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::vector< T > data( N * N );
+	std::fill(data.begin(), data.end(), static_cast< T >( 0 ) );
+	std::srand( RNDSEED );
+	for( size_t i = 0; i < N; ++i ) {
+		for( size_t j = i; j < N; ++j ) {
+			T val( std::rand(), std::rand() );
+			data[ i * N + j ] = val / std::abs( val );
+			data[ j * N + i ] += grb::utils::is_complex< T >::conjugate( data[ i * N + j ] );
+		}
+	}
+	return data;
+}
+
+//** generate upper/lower triangular part of a Symmetric matrix */
+template<
+	typename T
+>
+std::vector< T >  generate_symmherm_matrix_data(
+	size_t N,
+	const typename std::enable_if<
+		!grb::utils::is_complex< T >::value,
+		void
+	>::type * const = nullptr
+) {
+	std::vector< T > data( ( N * ( N + 1 ) ) / 2 );
+	std::srand( RNDSEED );
+	size_t k = 0;
+	for( size_t i = 0; i < N; ++i ) {
+		for( size_t j = i; j < N; ++j ) {
+			//data[ k ] = static_cast< T >( i + j*j ); // easily reproducible
+			data[ k ] = static_cast< T >( std::rand() )  / RAND_MAX;
+			++k;
+		}
+	}
+	return data;
+}
+
+//** check if rows/columns or matrix Q are orthogonal */
+template<
+	typename T,
+	typename Structure,
+	typename ViewType,
+	class Ring = Semiring< operators::add< T >, operators::mul< T >, identities::zero, identities::one >,
+	class Minus = operators::subtract< T >
+>
+RC check_overlap(
+	alp::Matrix< T, Structure, alp::Density::Dense, ViewType > &Q,
+	const Ring &ring = Ring(),
+	const Minus &minus = Minus()
+) {
+	const Scalar< T > zero( ring.template getZero< T >() );
+	const Scalar< T > one( ring.template getOne< T >() );
+
+	RC rc = SUCCESS;
+	const size_t n = nrows( Q );
+
+	// check if QxQt == I
+	alp::Matrix< T, Structure, alp::Density::Dense, ViewType > Qtmp( n );
+	rc = rc ? rc : set( Qtmp, zero );
+	rc = rc ? rc : mxm(
+		Qtmp,
+		Q,
+		conjugate( alp::get_view< alp::view::transpose >( Q ) ),
+		ring
+	);
+	Matrix< T, Structure, Dense > Identity( n );
```

Comment:
Shouldn't the Structure for identity matrix be a predefined one? At least Structure needs to be Symmetric, but this is not ensured.